core/app/models/spree/order_merger.rb

```diff
@@ -47,6 +47,7 @@ def handle_merge(current_line_item, other_order_line_item)
         handle_error(current_line_item) unless current_line_item.save
       else
         other_order_line_item.order_id = order.id
+        other_order_item.adjustments.update_all(order_id: order.id)
```

Comment:
@gloaec thank you for this PR, I believe there is a typo here - it should be `other_order_line_item`

Reply:
Indeed, sorry about that :)